front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -0,0 +1,212 @@
+      const dataSourceDescription = "Slack (managed)";
+      const dataSourceProviderId = "openai";
+      const dataSourceModelId = "text-embedding-ada-002";
+      const dataSourceMaxChunkSize = 256;
+
+      if (
+        !req.body.nangoConnectionId ||
+        typeof req.body.nangoConnectionId !== "string"
+      ) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The request body is invalid, expects \
+               { nangoConnectionId: string }.",
+          },
+        });
+      }
+
+      const dustProject = await DustAPI.createProject();
+      if (dustProject.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            // I don't know if we want to forward the core error to the client. LMK during code review please.
+            message: `Could not create the project. Reason: ${dustProject.error}`,
+          },
+        });
+      }
+
+      const providers = await Provider.findAll({
+        where: {
+          workspaceId: owner.id,
+        },
+      });
+      let credentials = credentialsFromProviders(providers);
+
+      const dustDataSource = await DustAPI.createDataSource(
+        dustProject.value.project.project_id.toString(),
+        {
+          dataSourceId: dataSourceName,
+          config: {
+            provider_id: dataSourceProviderId,
+            model_id: dataSourceModelId,
+            splitter_id: "base_v0",
+            max_chunk_size: dataSourceMaxChunkSize,
+            use_cache: false,
+          },
+          credentials,
+        }
+      );
+
+      if (dustDataSource.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "data_source_error",
+            message: `Could not create the data source. reason: ${dustDataSource.error}`,
+          },
+        });
+      }
+
+      const dataSource = await DataSource.create({
+        name: dataSourceName,
+        description: dataSourceDescription,
+        //assuming managed data sources are always private for now
+        visibility: "private",
+        config: JSON.stringify(dustDataSource.value.data_source.config),
+        dustAPIProjectId: dustProject.value.project.project_id.toString(),
+        workspaceId: owner.id,
+      });
+      const systemAPIKeyRes = await getSystemApiKey(owner.id);
+      if (systemAPIKeyRes.isErr()) {
+        console.error(
+          `Could not create the system API key: ${systemAPIKeyRes.error}`
+        );
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "Could not create the system API key",
+          },
+        });
+      }
+
+      try {
+        const connectorsRes = await fetch(
```

Comment:
We should go right away for a typed client. It's just a few lines of code and will help a lot the typing/error handling and readability of the code. (see lib/dust_api.ts)

Reply:
done.